### PR TITLE
feat: integrate Contact Us form into Contact Us page, ensure responsive layout, and add unit tests

### LINKED ISSUE #143: [Feature]: Develop Contact Us Page with Integrated Contact Us Form
### Description

Develop a Contact Us page with the already developed contact Us form previously developed by another intern. Ensure the page layout matches the provided design specifications. This page will feature the Contact Us form, developed by another intern, and ensure it functions correctly within the page layout and develop the content card on the right.

### Acceptance Criteria

- [ ] Integrate the Contact Us form component developed by another intern into the Contact Us page.
- [ ] Ensure the Contact Us form is displayed correctly with the content on the right within the page layout.
- [ ] Ensure the Contact Us page matches the provided design specifications and is responsive.
- [ ] Write unit tests to verify form validation, API integration, and responsiveness.

**Deadline: Saturday 12 Noon**
**Check #140 for the necessary component**
### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=432-7562&m=dev
image

### PR DESCRIPTION
**Pull request by ObiFaith**
# Description

Develop a Contact Us page with the already developed contact Us form previously developed by another intern. Ensure the page layout matches the provided design specifications. This page will feature the Contact Us form, developed by another intern, and ensure it functions correctly within the page layout and develop the content card on the right.

**Closes #143**

# Changes proposed

## What were you told to do?

- Used the layout component so I have both the Header and Footer in the Contact page
- Integrate the ContactForm component developed by another intern into the Contact page.
- Ensure the Contact Us form is displayed correctly with the content on the right within the page layout.
- Ensure the Contact Us page matches the provided design specifications and is responsive.
- Write unit tests to verify form validation, API integration, and responsiveness.

## What did you do?
- Added the icons that was used at the right side of the contact page in public/contact-icons
- Integrated ContactForm component which is expected on the left side (Issue 140)
- The Contact page can be found in src/app/contact
- I wrote the unit test to verify form validation, API integration, and responsiveness.
-  Made the Contact page to be responsive
- Used dummy data in /app/contact/constant.js for the right section of the Contact page

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos
Contact page UI
pnpm test report

### CODE REVIEW

File: src/app/contact/contact.test.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  fetch: typeof fetch;
+};
+
+describe("contact Page tests", () => {
+  // Create a fetch mock
+  const fetchMock = vi.fn();
+
+  beforeAll(() => {
+    global.fetch = fetchMock;
+  });
```
[the remark this answers is not in the document]
Comment (ObiFaith) [reply src/app/contact/contact.test.tsx]: Okay. I will fix that now

**Review by prudentbird — CHANGES_REQUESTED** [review]
Use lucide react icon library

File: public/images/newletter-email-template/Newsletter-Star.svg
Comment (prudentbird) [inline public/images/newletter-email-template/Newsletter-Star.svg:1]: Isn't this star in lucide react??

**Review by prudentbird — CHANGES_REQUESTED** [review]

File: public/images/newletter-email-template/Newsletter-Star.svg
Comment (prudentbird) [reply public/images/newletter-email-template/Newsletter-Star.svg:1]: @ObiFaith

Comment (ObiFaith) [reply public/images/newletter-email-template/Newsletter-Star.svg:1]: I don't think there is a section I am to use that star icon, you are suggesting. I worked on the right side section of the design and added components to complete the page while making the main section responsive

**Pull request closed without merging**